go/genesis/grpc.go

```diff
@@ -34,15 +32,9 @@ type grpcServer struct {
 // ToGenesis generates a genesis document based on current state at given height.
 func (s *grpcServer) ToGenesis(ctx context.Context, req *pb.GenesisRequest) (*pb.GenesisResponse, error) {
 	height := req.GetHeight()
-	if height <= 0 {
-		var err error
-		if height, err = s.tendermintService.GetHeight(); err != nil {
-			return nil, err
-		}
-	}
 
-	// Get genesis doc.
-	genesisProvider, err := New()
+	// Get tendermint state as a genesis doc.
```

Comment:
Since we are changing this, can we make it consensus backend-agnostic? E.g., make `consensus.Backend` provide some method that generates the genesis document and call it here instead of hardcoding "tendermint"?

Reply:
will do :+1:

Reply:
I moved it into tendermint/genesis/genesis.go for now and not yet part of the `consensus.Backend`. Added a comment on moving it into `consensus.Backend` interface with #1879